src/main/java/AddCommand.java

```diff
@@ -0,0 +1,88 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Class for the command that adds a task into the tasklist.
+ */
+public class AddCommand extends Command {
+
+    private String input;
+
+    /**
+     * Constructor for the AddCommand object.
+     *
+     * @param ipt the string representation of the command input.
+     */
+    public AddCommand(String ipt) {
+        input = ipt;
+    }
+
+    /**
+     * Indicates whether this is an exit command.
+     *
+     * @return false.
+     */
+    public boolean isExit() { return false; }
+
+    /**
+     * Executes the add command.
+     * The task will be parsed from the input string representation and added into the task list.
+     * The text file will be updated with the given task.
+     * A completion message will be displayed after successful execution.
+     *
+     * @param tasks tasklist that stores all the tasks
+     * @param ui ui object that deals with user interaction
+     * @param storage storage object that deals with writing into the text file
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        SimpleDateFormat format = new SimpleDateFormat("dd/MM/yyyy HHmm");
+        Task tk;
+        int idxOfSlash;
+
+        try {
+            if (input.length() >= 4) {
+
+                // add Deadline
+                if (input.length() >= 8 && input.substring(0, 8).equals("deadline")) {
```

Comment:
These conditions can work, but maybe you can try to use a .contains function?
for e.g.
input.contains("deadline")
This can simplify the entire conditional statement.